packages/mocha/src/index.ts

```diff
@@ -26,7 +26,7 @@ mocha.afterEach(async function() {
 
 function runInWorld(fn: TestFunction) {
   return async function(this: mocha.Context) {
-    await run((task) => fn.call(this, world!, task));
+    await run({ init: fn.bind(this) }, { resourceScope: world! });
```

Comment:
Why not just spawn this as a resource in world?

Reply:
I actually investigated this, it makes the error messages read a little bit worse, which is why I didn't do this at the moment. There might be ways around this, but this was the easier implementation.